pkg/evaluation/manager.go

```diff
@@ -0,0 +1,293 @@
+		if policyEntity == nil {
+			return nil, util.GrpcInternalError(log, "policy version does not exist", nil)
+		}
+
+		evaluatePolicyResponse, err := m.evaluatePolicy(ctx, policyAssignment.PolicyVersionId, policyEntity.RegoContent, occurrences)
+		if err != nil {
+			return nil, util.GrpcInternalError(log, fmt.Sprintf("error evaluating policy version %s", policyAssignment.PolicyVersionId), err)
+		}
+
+		if !evaluatePolicyResponse.Result.Pass {
+			resourceEvaluation.Pass = false
+		}
+
+		policyEvaluation := &pb.PolicyEvaluation{
+			Id:                   uuid.New().String(),
+			ResourceEvaluationId: resourceEvaluation.Id,
+			PolicyVersionId:      policyAssignment.PolicyVersionId,
+			Pass:                 evaluatePolicyResponse.Result.Pass,
+			Violations:           evaluatePolicyResponse.Result.Violations,
+		}
+
+		policyEvaluations = append(policyEvaluations, policyEvaluation)
+		bulkRequestItems = append(bulkRequestItems, &esutil.BulkRequestItem{
+			Operation:  esutil.BULK_CREATE,
+			Message:    policyEvaluation,
+			DocumentId: policyEvaluation.Id,
+			Join: &esutil.EsJoin{
+				Parent: resourceEvaluation.Id,
+				Field:  evaluationDocumentJoinField,
+				Name:   policyEvaluationRelationName,
+			},
+		})
+	}
+
+	response, err := m.esClient.Bulk(ctx, &esutil.BulkRequest{
+		Index: m.indexManager.AliasName(constants.EvaluationsDocumentKind, ""),
+		Items: bulkRequestItems,
+	})
+	if err != nil {
+		return nil, util.GrpcInternalError(log, "error storing resource evaluation results", err)
+	}
+	if err = util.CheckBulkResponseErrors(response); err != nil {
+		return nil, util.GrpcInternalError(log, "error storing resource evaluation results", err)
+	}
+
+	return &pb.ResourceEvaluationResult{
+		ResourceEvaluation: resourceEvaluation,
+		PolicyEvaluations:  policyEvaluations,
+	}, nil
+}
+
+func (m *manager) GetResourceEvaluation(ctx context.Context, request *pb.GetResourceEvaluationRequest) (*pb.ResourceEvaluationResult, error) {
+	panic("implement me")
+}
+
+func (m *manager) ListResourceEvaluations(ctx context.Context, request *pb.ListResourceEvaluationsRequest) (*pb.ListResourceEvaluationsResponse, error) {
+	panic("implement me")
+}
+
+func (m *manager) EvaluatePolicy(ctx context.Context, request *pb.EvaluatePolicyRequest) (*pb.EvaluatePolicyResponse, error) {
+	var err error
+	log := m.logger.Named("EvaluatePolicy").With(zap.String("policy", request.Policy), zap.String("resource", request.ResourceUri))
+	log.Debug("evaluate policy request received")
+
+	if request.ResourceUri == "" {
+		return nil, util.GrpcErrorWithCode(log, "resource uri is required", nil, codes.InvalidArgument)
+	}
+
+	policy, err := m.policyManager.GetPolicy(ctx, &pb.GetPolicyRequest{Id: request.Policy})
+	if err != nil {
+		return nil, util.GrpcInternalError(log, "error fetching policy", err)
+	}
+
+	// fetch occurrences from grafeas
+	occurrences, _, err := m.grafeasExtensions.ListVersionedResourceOccurrences(ctx, request.ResourceUri, "", constants.MaxPageSize)
```

Comment:
I know `EvaluatePolicy` doesn't do this at the moment either, but should this page through all occurrences? Or at least fail if the next page token is empty?

Reply:
The last time this came up we figured it would be best to log a warning here.  I can add that.